modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContext.java

```diff
@@ -1643,25 +1650,27 @@ private void _populateDDMDataProviderNavigationItem(
 	private List<DDMFormFieldType> _removeDDMFormFieldTypesOutOfScope(
 		List<DDMFormFieldType> ddmFormFieldTypes) {
 
-		List<DDMFormFieldType> availableDDMFormFieldTypes = new ArrayList<>();
+		List<String> ddmFormFieldTypesOutofScope = Arrays.asList(
+			DDMFormFieldTypeConstants.GEOLOCATION,
+			JournalArticleDDMFormFieldTypeConstants.JOURNAL_ARTICLE,
+			LayoutDDMFormFieldTypeConstants.LINK_TO_LAYOUT);
 
-		ListUtil.filter(
-			ddmFormFieldTypes, availableDDMFormFieldTypes,
-			ddmFormFieldType ->
-				!ddmFormFieldType.getName(
-				).equals(
-					"geolocation"
-				) &&
-				!ddmFormFieldType.getName(
-				).equals(
-					"journal_article"
-				) &&
-				!ddmFormFieldType.getName(
-				).equals(
-					"link_to_layout"
-				));
+		Stream<DDMFormFieldType> stream = ddmFormFieldTypes.stream();
 
-		return Collections.unmodifiableList(availableDDMFormFieldTypes);
+		return stream.filter(
+			ddmFormFieldType -> {
+				if (Objects.equals(
+						ddmFormFieldType.getName(),
+						DDMFormFieldTypeConstants.SEARCH_LOCATION)) {
+
+					return _ffSearchLocationDDMFormFieldTypeEnabled;
+				}
+
+				return !ddmFormFieldTypesOutofScope.contains(ddmFormFieldType);
```

Comment:
Hey, @marcelabc. I believe this was supposed to be `ddmFormFieldType.getName()`.